pkg/csi/service/common/vsphereutil.go

```diff
@@ -322,6 +358,45 @@ func CreateFileVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsCluster
 		}
 	}
 
+	if spec.StoragePolicyID != "" {
+
+		// Check storage policy compatibility.
+		var sharedDSMoRef []vim25types.ManagedObjectReference
+		for _, ds := range datastores {
+			sharedDSMoRef = append(sharedDSMoRef, ds.Reference())
+		}
+
+		compat, err := vc.PbmCheckCompatibility(ctx, sharedDSMoRef, spec.StoragePolicyID)
+		if err != nil {
+			return "", csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
```

Comment:
Is this a CSIInternalFault or can it classified as nonCSI fault ? Can you explain the corrective action for the users when they observe this error. I am guessing this is an incorrect configuration(user error) ?

Reply:
used newly introduced fault "CSIInvalidStoragePolicyConfigurationFault" instead of CSIInternalFault